src/main/java/org/opensearch/neuralsearch/processor/TextEmbeddingProcessor.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor;
+
+import java.util.*;
+import java.util.concurrent.ExecutionException;
+
+import lombok.extern.log4j.Log4j2;
+
+import org.apache.commons.lang3.StringUtils;
+import org.opensearch.ingest.AbstractProcessor;
+import org.opensearch.ingest.IngestDocument;
+import org.opensearch.neuralsearch.ml.MLCommonsClientAccessor;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * Before querying with neural search, all the documents should be ingested in the form of embedded vectors. Leaving the embedding process
+ * to the user offline will raise the bar of use, thus we create this processor for user input document text embedding.
+ */
+@Log4j2
+public class TextEmbeddingProcessor extends AbstractProcessor {
+
+    public static final String TYPE = "text_embedding";
+    public static final String MODEL_ID_FIELD = "model_id";
+    public static final String FIELD_MAP_FIELD = "field_map";
+
+    private static final String LIST_TYPE_NESTED_MAP_KEY = "knn";
+
+    private final String modelId;
+
+    private final Map<String, Object> fieldMap;
+
+    private final MLCommonsClientAccessor mlCommonsClientAccessor;
+
+    public TextEmbeddingProcessor(
+        String tag,
+        String description,
+        String modelId,
+        Map<String, Object> fieldMap,
+        MLCommonsClientAccessor clientAccessor
+    ) {
+        super(tag, description);
+        if (StringUtils.isBlank(modelId)) throw new IllegalArgumentException("model_id is null, can not process it");
+        if (fieldMap == null || fieldMap.size() == 0 || isEmbeddingConfigValid(fieldMap)) throw new IllegalArgumentException(
+            "Unable to create the TextEmbedding processor as field_map is null or empty."
+        );
+
+        this.modelId = modelId;
+        this.fieldMap = fieldMap;
+        this.mlCommonsClientAccessor = clientAccessor;
+    }
+
+    private boolean isEmbeddingConfigValid(Map<String, Object> fieldMap) {
+        return fieldMap.entrySet()
+            .stream()
+            .anyMatch(x -> StringUtils.isBlank(x.getKey()) || Objects.isNull(x.getValue()) || StringUtils.isBlank(x.getValue().toString()));
+    }
+
+    @Override
+    public IngestDocument execute(IngestDocument ingestDocument) {
+        validateEmbeddingFieldsType(ingestDocument, fieldMap);
+        Map<String, Object> knnMap = buildKnnMap(ingestDocument, fieldMap);
+        try {
+            List<List<Float>> vectors = mlCommonsClientAccessor.inferenceSentences(this.modelId, createInferenceList(knnMap));
+            processResponse(ingestDocument, knnMap, vectors);
+        } catch (ExecutionException | InterruptedException e) {
+            log.error("Text embedding processor failed with exception: " + e.getMessage(), e);
```

Comment:
```suggestion
            log.error("Text embedding processor failed with exception: ", e);
```
When exception trace is printed the exception message will be printed along that. So we can avoid doing e.getMessage()